server/channels/api4/custom_profile_attributes.go

```diff
@@ -192,13 +223,43 @@ func patchCPAValues(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var updates map[string]json.RawMessage
+	if err := json.NewDecoder(r.Body).Decode(&updates); err != nil {
+		c.SetInvalidParamWithErr("value", err)
+		return
+	}
+
 	auditRec := c.MakeAuditRecord("patchCPAValues", audit.Fail)
 	defer c.LogAuditRec(auditRec)
 	audit.AddEventParameter(auditRec, "user_id", userID)
 
-	results := make(map[string]string)
-	for fieldID, value := range attributeValues {
-		patchedValue, appErr := c.App.PatchCPAValue(userID, fieldID, strings.TrimSpace(value))
+	// Get all fields at once and build a map for quick lookup
+	allFields, appErr := c.App.ListCPAFields()
```

Comment:
The `App.PatchCPAValue` method is already fetching the fields and ensuring that they exist. Is there any reason to run the check at the API layer as well?

Reply:
I'm pulling the PropertyFields because to validate each PropertyValue, I need to know what its PropertyFields type is: if it's a simple `string` or a `user_id`, I can't validate them the same way so I need to know this info.

The validation of the PropertyValue could be moved somewhere else if needed (we tend to do validations in the store layer, which I always found super weird because we're allowing invalid info to flow quite deeply in our app - and also goes against all we learn in security training haha), but no matter where it goes, validating a PropertyValue will require knowing that its PropertyField's type is.

Reply:
Understood, and it makes sense. And I get your point about where to run validations layer-wise, and although I'm with you on this one, it's not something that we're going to solve as part of the PR :P so greenlight it is!